### DISCUSSION TITLE
Factory Methods That Allow For Intermediate Code To Execute And/Or Conditionally Fire An Observable When Bridging Between .NET Events and Observables

### DISCUSSION THREAD

**Discussion by nspoentgenIllumina** [reactiveui/ReactiveUI#3317] [Q&A] [answered]
Love the project! I see that Reactive Extensions + ReactiveUI give some factory methods to help with observable creation including bridging standard .NET events and observables. However, we sometimes need to do intermediate logic whenever an event fires and/or fire another event depending on some conditional statement. 

 Here's some example pseudocode of what our team would like to do:

event = standard .NET event

```
if (event0 || event1 || ... || eventN-1 || eventN) then:
    result = CustomFunction() //may or may not update class data
    if (CustomConditional(result)):
        fire observable
```

Is there a simple way to do this with existing factory methods?

**Comment by glennawatson** [comment, marked as answer]
I would checkout https://github.com/reactivemarbles/ObservableEvents for generating those observable events.

In terms of conditional, you can combine them together with `.CombineLatest(` or `Merge(` and then do a `.Where(` for your conditional logic.

**Reply by nspoentgenIllumina** [reply]
Ah, got it! The examples I saw online showed LINQ statements such as _Merge_ or _Where_ right before a termination operation such as _Subscribe_ or _ToProperty_, so I had it in my head that those methods were for consumers only. I didn't even realize that—except for termination methods—these methods are just returning new observables, so you can manipulate them at any point in the observer pipeline. This makes so much more sense now. Thank you!

FYI, I believe an example like this (taking an observable, manipulating it via LINQ, but _not_ immediately terminating it with a termination operator) would be beneficial in highlighting the composability of Reactive properties (maybe even a side-by-side with an _IEnumberable_ being converted to another _IEnumerable_ without using a _ToList_ or _ToArray_ to show the parallels). I know it seems so obvious in hindsight, but it can take quite a while to switch one's mind to the _Reactive_ way of thinking when one has done years and years of traditional event handling.

**Reply by glennawatson** [reply]
Think of observables as Linq for events. That helped my understanding a lot.
